core/src/instance.rs

```diff
@@ -349,6 +375,52 @@ pub mod tests {
         instance
     }
 
+    #[test]
+    /// This tests calling `process_action`
+    /// with an action that dispatches no new ones.
+    /// It tests that the desired effects do happen
+    /// to the state and that no observers or actions
+    /// are sent on the passed channels.
+    pub fn can_process_action() {
+        let mut instance = Instance::new();
+
+        let context = test_context("jane");
+        let (rx_action, rx_observer) = instance.initialize_channels();
+
+        let aw = test_action_wrapper_get();
+        let new_observers = instance.process_action(
```

Comment:
this is great @timotree3 

much nicer than having to run the loop and wait in tests. deterministic. no chance for a test to hang.